src/Nethermind/Nethermind.JsonRpc.Test/Data/Eip2930Tests.cs

```diff
@@ -0,0 +1,212 @@
+using Nethermind.Core.Test.Builders;
+using Nethermind.Int256;
+using Nethermind.JsonRpc.Data;
+using Nethermind.Serialization.Json;
+using NUnit.Framework;
+
+namespace Nethermind.JsonRpc.Test.Data
+{
+    public class Eip2930Tests
+    {
+        private readonly EthereumJsonSerializer _serializer = new EthereumJsonSerializer();
+        private Transaction _transaction = new Transaction();
+        private Dictionary<Address, IReadOnlySet<UInt256>> _data = new Dictionary<Address, IReadOnlySet<UInt256>>();
+        private TransactionForRpc _transactionForRpc = new TransactionForRpc();
+        private HashSet<UInt256> _storageKeys = new HashSet<UInt256>();
+
+
+        [Test]
+        public void can_serialize_valid_accessList()
+        {
+            _data = new Dictionary<Address, IReadOnlySet<UInt256>>();
+            _transaction = new Transaction();
+            _transaction.Type = TxType.AccessList;
+
+            _storageKeys = new HashSet<UInt256> {1, 2, 3, 5, 8};
+            _data.Add(TestItem.AddressA, _storageKeys);
+            _storageKeys = new HashSet<UInt256> {42};
+            _data.Add(TestItem.AddressB, _storageKeys);
+            _transaction.AccessList = new AccessList(_data);
+            _transactionForRpc = new TransactionForRpc(_transaction);
+
+            string serialized = _serializer.Serialize(_transactionForRpc);
+            string expected = "{\"nonce\":\"0x0\",\"blockHash\":null,\"blockNumber\":null,\"transactionIndex\":null,\"to\":null,\"value\":\"0x0\",\"gasPrice\":\"0x0\",\"gas\":\"0x0\",\"input\":null,\"type\":\"0x01\",\"accessList\":[{\"address\":\"0xb7705ae4c6f81b66cdb323c65f4e8133690fc099\",\"storageKeys\":[\"0x1\",\"0x2\",\"0x3\",\"0x5\",\"0x8\"]},{\"address\":\"0x942921b14f1b1c385cd7e0cc2ef7abe5598c8358\",\"storageKeys\":[\"0x2a\"]}]}";
+            expected.Should().Be(serialized);
+        }
+
+        [Test]
+        public void can_deserialize_valid_accessList()
+        {
+            _transactionForRpc = _serializer.Deserialize<TransactionForRpc>("{\"nonce\":\"0x0\",\"blockHash\":null,\"blockNumber\":null,\"transactionIndex\":null,\"to\":null,\"value\":\"0x0\",\"gasPrice\":\"0x0\",\"gas\":\"0x0\",\"input\":null,\"type\":\"0x01\",\"accessList\":[{\"address\":\"0xb7705ae4c6f81b66cdb323c65f4e8133690fc099\",\"storageKeys\":[\"0x1\",\"0x2\",\"0x3\",\"0x5\",\"0x8\"]},{\"address\":\"0x942921b14f1b1c385cd7e0cc2ef7abe5598c8358\",\"storageKeys\":[\"0x2a\"]}]}");
+
+            _transactionForRpc.Type.Should().Be(1);
+            _transactionForRpc.AccessList.Length.Should().Be(2);
+            _transactionForRpc.AccessList[0].Address.Should().Be(TestItem.AddressA);
+            _transactionForRpc.AccessList[0].StorageKeys.Should().BeEquivalentTo(new HashSet<UInt256>{1, 2, 3, 5, 8});
+            _transactionForRpc.AccessList[1].Address.Should().Be(TestItem.AddressB);
+            _transactionForRpc.AccessList[1].StorageKeys.Should().BeEquivalentTo(new HashSet<UInt256> {42});
```

Comment:
Can you use should().BeEquivalentTo() and compare to fully constructed object graph?